### ISSUE TITLE
Vehicle disappears on explode sometimes

### ISSUE THREAD

**Issue by rt-2** [Open-GTO/sa-mp-fixes#60]
Hi, I am not sure if this was addressed already.
When a vehicle explode, there is a chance that it doesn't re-spawn.
I am not sure if it is related to the fact that no one would never have been in the vehicle yet when it explodes.
I know this issue is known in the community, do you guys know what I'm talking about?
Thank you,
rt-2

**Comment by IstuntmanI** [comment]
Yeah, I get this bug sometimes in my server. AFAIK, it happens indeed when the vehicle explodes and it didn't have any driver. I didn't research this more. It's a good thing that it is pretty hard to explode vehicles that weren't occupied, the only possible way to do this AFAIK is only be making them stand upside down.

If a fix could be made, we would have to use ```OnVehicleDeath``` with a check for the vehicle health and if it was ever occupied, then use ```SetVehicleToRespawn```.

Not sure if it is related to the respawn delay too.

EDIT: Despite the title, it isn't disappearing, but instead it won't ever respawn, it will just sit there exploded if it is streamed-in to you, or it will be on fire and explode over and over again if you restream that vehicle.

**Comment by rt-2** [comment]
For me, it becomes invisible to the player who streamed it explode. Then if I reconnect I can re-stream it and it re-explode.
My problem I think is that it re-spawns it client side, but the server see nothing. So it doesn't call the re-spawn functions, health is not re-initialized, so the vehicle keep exploding.
I think we'd need to force re-spawn yes.
rt-2

**Comment by rt-2** [comment]
What I tested:
Both `OnVehicleDeath` and `OnVehicleSpawn` aren't called when some players are streaming the vehicle exploding, but none of them ever was in the vehicle. Even if one of them witnessed someone in the vehicle's driver position, it needs to be themselves. And it needs to be the driver's seat, it doesn't work with passengers.
I tested this and came up with those results. I am 99% sure this is all the parameters to this problem.
Now we just need to solve it :P.
I have an idea where we could check if a players have been in the driver seat of a vehicle, then set this in a variable that get cleared everytime the player unstream the vehicle. The big problem is we need a way to know when it explodes. Then we check is any player that is streaming atm have been driver of that vehicle and call the missing functions accordingly.
At the moment I cannot see any place to see that, `OnVehicleDamageStatusUpdate` "does not include vehicle health changes". I hope we can find a better way than a loops in `OnPlayerUpdate`.
rt-2
EDIT: I have not played with the respawnDelay and I used this function `AddStaticVehicleEx(vehid, x, y, z, angle, color1, color2, -1);`.

**Comment by IstuntmanI** [comment]
It probably calls ```OnUnoccupiedVehicleUpdate``` instead when it explodes, we could check its health. If it is ```0.0``` (the vehicle health is ```0.0``` only when it exploded I think, not sure if it is possible to reach ```0.0``` if damaging it while it is on fire) and it wasn't ever occupied by a driver, it should call ```SetVehicleToRespawn```.

**Comment by rt-2** [comment]
Unfortunately that doesn't look like a doable thing. I tried testing with setting the vehicle's health to 100, well I do receive a bunch of `OnUnoccupiedVehicleUpdate`, but the health I get with `GetVehicleHealth` is 100 each time, same when I put 0.1 or 10.
Thank you,
rt-2

**Comment by rt-2** [comment]
I normally test a lot of my new script in filterscript, this only works int he gamemode
